axelor-purchase/src/main/java/com/axelor/apps/purchase/service/PurchaseRequestServiceImpl.java

```diff
@@ -53,4 +68,104 @@ public void acceptRequest(List<PurchaseRequest> purchaseRequests) {
       purchaseRequestRepo.save(purchaseRequest);
     }
   }
+
+  @Transactional
+  @Override
+  public ActionViewBuilder generatePo(
```

Comment:
Better return PO list and do ActionViewBuilder on controller, so that this method can be used by other service/method.